### PR TITLE
Fixes #22969: Allow tasks to use Org.current

### PR DESCRIPTION
**Pull request by akofink**
Basically copied verbatim from https://github.com/theforeman/foreman-tasks/blob/master/app/lib/actions/middleware/keep_current_user.rb

### CODE REVIEW

File: app/lib/actions/middleware/keep_current_organization.rb [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+# 2 of the License (GPLv2) or (at your option) any later version.
+# There is NO WARRANTY for this software, express or implied,
+# including the implied warranties of MERCHANTABILITY,
+# NON-INFRINGEMENT, or FITNESS FOR A PARTICULAR PURPOSE. You should
+# have received a copy of GPLv2 along with this software; if not, see
+# http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+
+module Actions
+  module Middleware
+    class KeepCurrentOrganization < Dynflow::Middleware
```
Comment (iNecas) [inline app/lib/actions/middleware/keep_current_organization.rb]: What about using `KeepCurrentTaxonomies` and storing both org and location in the context?

**Comment by iNecas** [comment]
Maybe this request will be a bit bold, but would you mind adding some test case (inspiration can be taken from https://github.com/theforeman/foreman-tasks/blob/master/test/unit/actions/proxy_action_test.rb) to actually cover the behavior? Bonus points for adding that also for `KeepCurrentUser`, which I guess would be quite simple to do, once this is done. And sorry for not enforcing this in the past, I realize it would make the life easier for folks now.

**Comment by akofink** [comment]
Thanks for checking me. :) How could I submit a PR without tests! :see_no_evil:

**Comment by akofink** [comment]
This PR is blocking some other work in Katello. Would you mind if I add middleware tests in a separate PR?

**Comment by iNecas** [comment]
Ok, setting recurring reminder to ping you to add the tests later :) Merged as 8e9e7daaac397a4a6dca3c91a046926dc6fbae42,  foreman-tasks 0.12.2 pushed to rubygems

**Comment by mmoll** [comment]
```
app/lib/actions/middleware/keep_current_taxonomies.rb:46:11: C: Naming/PredicateName: Rename has_current_taxonomies? to current_taxonomies?.
    def has_current_taxonomies?
        ^^^^^^^^^^^^^^^^^^^^^^^
app/lib/actions/middleware/keep_current_taxonomies.rb:48:9: C: Layout/MultilineOperationIndentation: Use 2 (not 0) spaces for indenting an expression spanning multiple lines.
    (Location.current || action.input[:current_location_id].nil?)
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

**Comment by akofink** [comment]
Huh - I guess we should run rubocop on PRs in this repo.

**Comment by iNecas** [comment]
Lessons learned, let's address that in https://github.com/theforeman/foreman-tasks/pull/325

**Pull request closed without merging**